modules/core/src/main/java/org/apache/ignite/internal/IgniteMessagingImpl.java

```diff
@@ -207,7 +210,7 @@ private void send0(@Nullable Object topic, Collection<?> msgs, boolean async) th
         guard();
 
         try {
-            GridContinuousHandler hnd = new GridMessageListenHandler(topic, (IgniteBiPredicate<UUID, Object>)p);
+            GridContinuousHandler hnd = new GridMessageListenHandler(topic, securityAwareBiPredicate(p));
```

Comment:
Do we should add predicate to async implementation `remoteListenAsync` ?

Reply:
yes!